### PR TITLE
feat: introduce channels

### LINKED ISSUE #325: Implement get joined channels/friends
- Implementation in friends saga
- It should be paginated
- Message from renderer to kernel with # page, # results, type (CHANNEL or FRIEND/PRIVATE)
- Result of get joined from kernel to renderer

### LINKED ISSUE #330: Implement join channel
- [x] friends saga: implement join channel function
- [x] chat saga: /join chat command with channel name
- [x] unity->kernel message: join channel with channel name
- [x] kernel->unity message: confirm or err on join channel

### LINKED ISSUE #468: Implement get channel messages window
- [x] Mark messages as seen
- [x] Update total messages by channel and total messages
- [x] Get messages window

### PR DESCRIPTION
**Pull request by Julieta11**
Closes #325 
Closes #330 
Closes #468 

# What? 
Base branch for channels feature. Find more info [here](https://www.notion.so/decentraland/RFC-Technical-proposal-for-Channels-8fdbd914091f4d2194424caf9e3e096b).

So far, the PR addresses:
- join or create channel via command,
- create channel,
- ability to get all channels where the user is joined,
- and show channel messages window.

### CODE REVIEW

**Comment by guidota** [comment]
Should we track events like channel creation, joins, leaves, etc? I think those metrics would be very helpful to understand how the feature is being used.

File: packages/shared/friends/sagas.ts [hunk 1]
```diff
@@ -1179,3 +1193,241 @@ function updateSocialInfo(socialData: SocialData) {
     })
   )
 }
+
+// Join or create channel
+async function* handleJoinOrCreateChannel(action: JoinOrCreateChannel) {
```
Comment (menduz) [inline packages/shared/friends/sagas.ts:1198]: Sagas can't be async
```suggestion
function* handleJoinOrCreateChannel(action: JoinOrCreateChannel) {
```

File: packages/shared/friends/sagas.ts [hunk 2]
```diff
@@ -1179,3 +1193,241 @@ function updateSocialInfo(socialData: SocialData) {
...
+  try {
+    const client: SocialAPI | null = getSocialClient(store.getState())
+    if (!client) return
+
+    const channelId = action.payload.channelId
+    const ownId = client.getUserId()
+
+    // get or create channel
+    const { created, conversation } = await client.getOrCreateChannel(channelId, [ownId])
```
Comment (menduz) [inline packages/shared/friends/sagas.ts:1207]: Convert async calls this way
```suggestion
    const { created, conversation } = yield apply(client, client.getOrCreateChannel, [channelId, [ownId]])
```

File: packages/unity-interface/BrowserInterface.ts [hunk 3]
```diff
@@ -686,6 +697,57 @@ export class BrowserInterface {
...
+      defaultLogger.error('error createChannel', err),
+        trackEvent('error', {
+          message: `error creating channel ${createChannelPayload.channelId} ` + err.message,
+          context: 'kernel#friendsSaga',
+          stack: 'createChannel'
+        })
+    })
+  }
+
+  public async MarkChannelMessagesAsSeen(markChannelMessagesAsSeenPayload: MarkChannelMessagesAsSeenPayload) {
```
Comment (menduz) [inline packages/unity-interface/BrowserInterface.ts:711]:
```suggestion
  public MarkChannelMessagesAsSeen(markChannelMessagesAsSeenPayload: MarkChannelMessagesAsSeenPayload) {
```

File: packages/unity-interface/BrowserInterface.ts [hunk 4]
```diff
@@ -686,6 +697,57 @@ export class BrowserInterface {
     }
   }
 
+  public async CreateChannel(createChannelPayload: CreateChannelPayload) {
```
Comment (menduz) [inline packages/unity-interface/BrowserInterface.ts:700]: No need for async when using .catch
```suggestion
  public CreateChannel(createChannelPayload: CreateChannelPayload) {
```

**Review by menduz — APPROVED** [review]
Code looking good, some actions are commented like leaveChannel. If that's intended we are go to merge

**Pull request merged**